openfisca_france_local/departements/haute_saone/eligibilite.py

```diff
@@ -0,0 +1,13 @@
+from openfisca_france.model.base import Variable, Menage, MONTH
+
+from numpy.core.defchararray import startswith
+
+
+class haute_saone_eligibilite_residence(Variable):
+    value_type = bool
+    entity = Menage
+    definition_period = MONTH
+    label = "Éligibilité résidentielle d'un ménage aux dipositifs de la Haute-Saone"
+
+    def formula(menage, period):
+        return startswith(menage('depcom', period), str.encode('70'))
```

Comment:
L'encodage ne me semble pas nécessaire pour cette chaîne
```suggestion
        return startswith(menage('depcom', period), 70')
```

Reply:
Apparement on peut aussi optimiser les perfs avec `char.startswith` de numpy (à vérifier) :

```python
  import numpy as np

  ...

  return np.char.startswith(...
```

Reply:
> 

Ici, `menage('depcom', period)` retourne un tableau de `bytes` qu'on ne peut comparer directement avec un `string` ou un `number` (enfin, techniquement Python nous laisse le faire mais le comportement n'est pas celui qui nous interesse, ou alors il nous manque quelque chose pour que ça corresponde à ce qu'on veut). 
On utilise `str.encore()` pour convertir le numéro de département en byte et ensuite le comparer.

Reply:
> Apparement on peut aussi optimiser les perfs avec `char.startswith` de numpy (à vérifier; ma propositition est issue d'une version plus récente) :
> 
> ```python
>   import numpy as np
> 
>   ...
> 
>   return np.char.startswith(...
> ```

J'ai passé un peu de temps à enquêter sur ton retour, d'après la doc de Numpy, il y a en effet d'autres entrées quand on cherche `startswith` dans le moteur de recherche.

Je suis plutôt confus dans tout ça, de ce que je comprends il ya 4 import différent pour faire la même chose : 
```python
        char.chararray.startswith
        char.startswith
        chararray.startswith
        char.chararray.startswith
```

Tu proposes ` np.char.startswith`, or ça fait référence à la même fonction que celle qu'on utilise déjà (par un jeu d'import)

Les 3 autres : `char.chararray.startswith`, `chararray.startswith`, `char.chararray.startswith` font référence à une autre fonction mais ils la partagent.
Là où ça me dépasse c'est que ce sont des méthodes d'une class `class chararray(ndarray[_ShapeType, _CharDType]):`
surchargée.

Toutes fonctionnent mais je n'arrive pas à voir les différences ou les intérêt. Je ne vois rien dans la doc

Pourrais-tu élaborer stp ? @Shamzic

Reply:
> Ici, `menage('depcom', period)` retourne un tableau de `bytes` qu'on ne peut comparer directement avec un `string` ou un `number` (enfin, techniquement Python nous laisse le faire mais le comportement n'est pas celui qui nous interesse, ou alors il nous manque quelque chose pour que ça corresponde à ce qu'on veut). On utilise `str.encore()` pour convertir le numéro de département en byte et ensuite le comparer.

Ok! Je pensais que cela renvoyait un `string`, pourrais-tu m'indiquer où est-ce qu'on peut trouver cette information s'il te plaît ?

Reply:
Concernant `startswith`, c'est l'IA qui m'a proposé cela en suggérant une amélioration de performances (d'où ma proposition "à vérifier") et après véirification, c'est effectivement mieux de laisser l'import spécifique plutôt que d'importer tout le module `numpy` comme suggéré au départ.

Reply:
> > Ici, `menage('depcom', period)` retourne un tableau de `bytes` qu'on ne peut comparer directement avec un `string` ou un `number` (enfin, techniquement Python nous laisse le faire mais le comportement n'est pas celui qui nous interesse, ou alors il nous manque quelque chose pour que ça corresponde à ce qu'on veut). On utilise `str.encore()` pour convertir le numéro de département en byte et ensuite le comparer.
> 
> Ok! Je pensais que cela renvoyait un `string`, pourrais-tu m'indiquer où est-ce qu'on peut trouver cette information s'il te plaît ?

Alors là... c'est un grand mystère pour moi! Je le sais parce que c'est ce que je vois quand je manipule ces données mais la magie qui explique le pourquoi ce sont des `bytes` est un peu trop dark pour moi. J'ai passé un moment à chercher :/ Désolé @Shamzic

Reply:
Pas de soucis ^^